src/storage/txn/commands/check_txn_status.rs

```diff
@@ -58,6 +58,12 @@ impl<S: Snapshot, L: LockManager> WriteCommand<S, L> for CheckTxnStatus {
     /// the `current_ts` is literally the timestamp when this function is invoked. It may not be
     /// accurate.
     fn process_write(mut self, snapshot: S, context: WriteContext<'_, L>) -> Result<WriteResult> {
+        let new_max_ts = cmp::max(
+            cmp::max(self.lock_ts, self.caller_start_ts),
+            self.current_ts,
```

Comment:
`current_ts` may be max uint64.

Reply:
good catch...

Reply:
It's `caller_start_ts` that may be max uint64.
`current_ts` must be greater than a non-max `caller_start_ts`, so I just don't update max_ts with `caller_start_ts`.